src/Mod/Draft/draftgeoutils/geometry.py

```diff
@@ -275,32 +262,119 @@ def getRotation(v1, v2=App.Vector(0, 0, 1)):
     return App.Rotation(axis, angle)
 
 
-def isPlanar(shape):
+def isPlanar(shape, tol=-1):
     """Return True if the given shape or list of points is planar."""
-    n = getNormal(shape)
-    if not n:
-        return False
 
+    # for points    
     if isinstance(shape, list):
         if len(shape) <= 3:
             return True
         else:
-            for v in shape[3:]:
-                pv = v.sub(shape[0])
-                rv = DraftVecUtils.project(pv, n)
-                if not DraftVecUtils.isNull(rv):
-                    return False
+            poly = Part.makePolygon(shape)
+            if isStraightLine(poly, tol):
+                return True
+            plane = poly.findPlane(tol)
+            if plane:
+                return True
+            else:
+                return False
+    
+    # for shapes
+    # because Part.Shape.findPlane return None for Vertex and straight edges
+    if shape.ShapeType == "Vertex":
+        return True
+
+    if isStraightLine(shape, tol):
+        return True
+
+    plane = shape.findPlane(tol)
+    if plane:
+        return True
     else:
-        if len(shape.Vertexes) <= 3:
-            return True
+        return False
+
+
+def isStraightLine(shape, tol=-1):
+    """Return True if shape is a straight line."""
+    # function used in other methods because Part.Shape.findPlane assign a
+    # plane and normal to straight wires creating priviliged directions
+    # and to deal with straight wires with overlapped edges 
```

Comment:
You can add this text to the documentation string. The more documentation we provide the better.
```py
def is_straight_line(shape, tol=-1):
    """Return True if shape is a straight line.

    This function is used in other methods because `Part.Shape.findPlane` assigns a
    plane and normal to straight wires, creating privileged directions,
    and to deal with straight wires with overlapped edges.
    """
```

I'm not sure what you mean by "privileged directions". Is this a mathematical term?